src/boot/efi/random-seed.c

```diff
@@ -116,20 +116,36 @@ static void validate_sha256(void) {
 #endif
 }
 
+typedef struct {
+        void *p;
+        size_t size;
+} VarEraser;
+
+static void erase_var(VarEraser *e) {
+        explicit_bzero_safe(e->p, e->size);
+}
+
+#define CLEANUP_ERASE(obj) \
+        _cleanup_(erase_var) _unused_ VarEraser UNIQ_T(eraser, UNIQ) = { .p = &obj, .size = sizeof(obj) }
```

Comment:
hmm, the uniq stuff isn't really necessary here, is it? i mean, just suffix a fixed string would suffice, and has the nice benefit that you might not accidentially defined two erasers for the same field.